tests/main/test_mp.py

```diff
@@ -115,38 +114,92 @@ def test_mp(tmp_path: Path):
 
     with tempfile.TemporaryDirectory() as fp:
         config.experiment_dir = fp
-        assert_error_msg(
-            lambda: ParallelTrainer(wrapper=wrapper, run_config=config),
-            "Device must be set to 'cuda' if `gpu_mb_per_experiment` > 0",
-        )
         config.device = "cuda"
-        config.gpu_mb_per_experiment = 1e12
-        config.cpus_per_experiment = 1e12
-        out, err = capture_output(
-            lambda: ParallelTrainer(wrapper=wrapper, run_config=config)
-        )
-        assert (
-            "Expected GPU memory utilization" in out
-            and "Expected CPU core util. exceed system capacity" in out
-        )
-
+        if torch.cuda.is_available():
+            config.gpu_mb_per_experiment = 1e12
+            config.cpus_per_experiment = 1e12
+            out, err = capture_output(
+                lambda: ParallelTrainer(wrapper=wrapper, run_config=config)
+            )
+            assert (
+                    "Expected GPU memory utilization" in out
+                    and "Expected CPU core util. exceed system capacity" in out
+            )
+        else:
+            assert_error_msg(
+                lambda: ParallelTrainer(wrapper=wrapper, run_config=config),
+                "Could not find a torch.cuda installation on your system."
```

Comment:
We removed the cuda requirement why is this test passing?

Reply:
Since there's no cuda device in container, the error "Could not find a torch.cuda installation on your system." when initializing the trainer with `config.device="cuda"` will be raised